src/components/selectable/selectable.tsx

```diff
@@ -611,60 +637,118 @@ export class EuiSelectable<T = {}> extends Component<
       </EuiI18n>
     ) : undefined;
 
-    const listAccessibleName = getAccessibleName(listProps);
+    let listScreenReaderStatus: ReactNode = null;
+    const resultsLength = visibleOptions.filter((option) => !option.disabled)
+      .length;
+    if (resultsLength === 0) {
+      listScreenReaderStatus = (
+        <EuiI18n
+          token="euiSelectable.noSearchResults"
+          default="No search results"
+        />
+      );
+    } else if (resultsLength === 1) {
+      listScreenReaderStatus = (
+        <EuiI18n
+          token="euiSelectable.singleSearchResult"
+          default="1 result available"
+        />
+      );
+    } else {
+      listScreenReaderStatus = (
+        <EuiI18n
+          token="euiSelectable.multipleSearchResults"
+          default="{resultsLength} results available"
+          values={{
+            resultsLength,
+          }}
+        />
+      );
+    }
```

Comment:
I think we should simplify this to not split up multiple vs singular pluralization. For one, in some languages pluralization isn't a thing, so they would have no need for the extra i18n token, and for another, other i18n frameworks (e.g. the one Kibana uses) have the ability to handle pluralization baked into their strings, and they have no need to have separate tokens/IDs for pluralized strings either.

It looks like we've manually handled pluralization via conditional logic in other places:

https://github.com/elastic/eui/blob/1999ef6cf52480bf159a5238d89db542176af28b/src/components/datagrid/column_sorting.tsx#L66-L75

But personally I think the good old `(s)` option is fine as a simpler way of handling this. Again, in Kibana what should happen is that the token will get correctly pluralized via their i18n'ing of our tokens, so this shouldn't be an issue on actual production. 

```suggestion
    } else {
      listScreenReaderStatus = (
        <EuiI18n
          token="euiSelectable.searchResults"
          default="{resultsLength} result(s) available"
          values={{ resultsLength }}
        />
      );
    }
```